### PR TITLE
watchdog: Watchdog Callback, main MCU

### PR DESCRIPTION
**Pull request by sri9311**
callback functionality added to watchdog

### CODE REVIEW

File: lib/errors/errors_tests.c [hunk 1]
```diff
@@ -153,7 +153,7 @@ fatal_errors_trigger(enum error_case_e type)
 #endif
 #ifdef CONFIG_ORB_LIB_WATCHDOG
     case FATAL_WATCHDOG:
-        (void)watchdog_init();
+        (void)watchdog_init(NULL);
```
Comment (fouge) [inline lib/errors/errors_tests.c:156]: Can you check that this triggers a watchdog reset? 🤔 

I am not sure it did. And it's very likely that this won't fix it.
Make sure to pass a callback that returns `false`

**Review by fouge — CHANGES_REQUESTED** [review]

**Review by fouge — CHANGES_REQUESTED** [review]
also make sure to install the pre-commit hooks and run the formatter

File: lib/watchdog/watchdog.c [hunk 2]
```diff
@@ -27,16 +29,24 @@ static void
...
     LOG_ERR("Watchdog thread stopped");
 }
 
 int
-watchdog_init(void)
+watchdog_init(bool (*callback)(void))
 {
+    if (callback == NULL) {
+        return RET_ERROR_INVALID_PARAM;
+    }
```
[the remark this answers is not in the document]
Comment (sri9311) [reply lib/watchdog/watchdog.c:46]: I think a callback should always be required, that way the stub always exists. in this case it'll always return true

File: lib/errors/errors_tests.c (hunk 1 above)
Comment (sri9311) [reply lib/errors/errors_tests.c:156]: will check this

**Review by fouge — CHANGES_REQUESTED** [review]

**Review by fouge — APPROVED** [review]

**Pull request merged**